src/Workspaces/Core/Portable/Classification/ClassifierHelper.cs

```diff
@@ -45,12 +46,15 @@ internal static partial class ClassifierHelper
             using var _1 = Classifier.GetPooledList(out var syntaxSpans);
             using var _2 = Classifier.GetPooledList(out var semanticSpans);
 
-            await classificationService.AddSyntacticClassificationsAsync(document, span, syntaxSpans, cancellationToken).ConfigureAwait(false);
+            for (var i = 0; i < spans.Length; i++)
```

Comment:
>for (var i = 0; i < spans.Length; i++) [](http://example.com/codeflow?start=12&length=38)

Did this one not lend itself to passing down the array like the others?

Reply:
i would pass this along.  The *inner* call can do thsi loop.